src/idl_gen_cpp.cpp

```diff
@@ -317,6 +317,11 @@ class CppGenerator : public BaseGenerator {
 
     if (opts_.gen_nullable) { code_ += "#pragma clang system_header\n\n"; }
 
+    if (opts_.g_cpp_std >= cpp::CPP_STD_17) {
```

Comment:
should this instead go into `base.h` or similar?

Reply:
I think I'd probably avoid that, because the `<tuple>` header is a bit long and template heavy, and so we probably don't want to force users to include it unless they have this C++ static reflection feature enabled (with C++17).